src/transforms/regex_parser.rs

```diff
@@ -143,36 +207,20 @@ impl RegexParser {
         overwrite_target: bool,
         types: HashMap<Atom, Conversion>,
     ) -> Self {
-        // Build a buffer of the regex capture locations to avoid
+        // Build a buffer of the regex capture locations and names to avoid
         // repeated allocations.
-        let capture_locs = patterns
-            .iter()
-            .map(|regex| regex.capture_locations())
-            .collect();
-
-        // Calculate the location (index into the capture locations) of
-        // each named capture, and the required type coercion.
-        let capture_names: Vec<(usize, Atom, Conversion)> = patterns
-            .iter()
-            .map(|regex| {
-                let capture_names: Vec<(usize, Atom, Conversion)> = regex
-                    .capture_names()
-                    .enumerate()
-                    .filter_map(|(idx, cn)| {
-                        cn.map(|cn| {
-                            let cn: Atom = cn.into();
-                            let conv = types.get(&cn).unwrap_or(&Conversion::Bytes);
-                            (idx, cn, conv.clone())
-                        })
-                    })
-                    .collect();
-                capture_names
-            })
-            .flatten()
+        let patterns: Vec<CompiledRegex> = patterns
+            .into_iter()
+            .map(|regex| CompiledRegex::new(regex, &types))
             .collect();
 
         // Pre-calculate if the source field name should be dropped.
-        drop_field = drop_field && !capture_names.iter().any(|(_, f, _)| *f == field);
+        drop_field = drop_field
+            && !patterns
+                .iter()
+                .map(|p| &p.capture_names)
+                .flatten()
+                .any(|(_, f, _)| *f == field);
```

Comment:
Doesn't need to be addressed in this PR, but this is making me wonder if we should take into account which regex matched when deciding whether or not to drop the source field. Right now we'll do it consistently, which is good, but there may be use cases where varying based on the capture names of the specific regex would be desirable.